lib/translator/read.cc

```diff
@@ -194,4 +194,13 @@ StatusCode Read16ToNvme(Span<const uint8_t> scsi_cmd,
               nvme_cmd, allocation, nsid);
 }
 
+void ReadToScsi(Span<uint8_t> buffer,
+                const nvme::GenericQueueEntryCmd& nvme_cmd) {
+  uint64_t data_ptr = nvme_cmd.dptr.prp.prp1;
+  // cdw 12 nlb bits 15:00 (zero based field)
+  uint16_t transfer_length = nvme_cmd.cdw[2] + 1;
+
+  memcpy(buffer.data(), reinterpret_cast<uint8_t*>(data_ptr), transfer_length);
```

Comment:
readval

Reply:
Can't really use ReadVal here because I need to specifically copy `transfer_length` number of bytes :(